### ISSUE TITLE
Feature: Text to speech

### ISSUE THREAD

**Issue by CaptainCalliope** [logancyang/obsidian-copilot#67]
It would be useful to have chat output come out as text to speech, with a number of text to speech options including default OS TTS and integration with ElevenLabs and other text to speech LMs.

Being able to highlight text and have it read out loud would also be super helpful, as well as being able to save recordings and embed them in documents.

**Comment by logancyang** [comment]
That's an interesting thought. Not sure how it works with the plugin ecosystem yet, but will find out.

**Comment by CaptainCalliope** [comment]
For context, [this browser extension](https://github.com/C-Nedelcu/talk-to-chatgpt) is what made me really want this. Not saying this is the ideal UX, but it should give you an idea as to what kind of experience is possible.

**Comment by albirrkarim** [comment]
Hi @logancyang @CaptainCalliope 

Just to let you know, I created a library for it. But unfortunately it is not available as open source. This library can be useful when you are creating products for clients who buy.

[![React / Vanilla Speech Highlight](https://raw.githubusercontent.com/albirrkarim/react-speech-highlight-demo/main/img/banner.png)](https://react-speech-highlight.vercel.app)

React / Vanilla Speech Highlight is a powerful library for integrating text-to-speech and real-time word/sentence highlighting into your web applications. It supports audio files, the Text-to-Speech API, and the Web Speech Synthesis API, making it ideal for creating interactive, accessible, and dynamic user experiences.

Check out my research: https://github.com/albirrkarim/react-speech-highlight-demo
Try live demo: https://react-speech-highlight.vercel.app

**Comment by logancyang** [comment]
Thanks for reporting this, and sorry for the slow reply.

Copilot v3 is in code freeze, so we're closing this as not planned. v4 is a ground-up rewrite and is launching soon, with a very different architecture around agent mode.

If this is still an issue for you in v4 once it lands, please open a new issue and we'll take a fresh look. 🙏
